hack/unwanted-dependencies.json

```diff
@@ -180,21 +148,51 @@
       "go.opencensus.io": [
         "github.com/Microsoft/hcsshim"
       ],
+      "google.golang.org/appengine": [
+        "github.com/prometheus/client_golang"
+      ],
       "golang.org/x/exp": [
+        "github.com/Microsoft/hcsshim",
         "github.com/antlr4-go/antlr/v4",
+        "github.com/cilium/ebpf",
+        "github.com/containerd/cgroups/v3",
         "github.com/google/cel-go",
         "k8s.io/mount-utils"
       ],
-      "google.golang.org/appengine": [
-        "github.com/prometheus/client_golang"
+      "github.com/gorilla/mux": [
+        "github.com/Microsoft/hcsshim"
       ],
       "google.golang.org/genproto": [
-        "github.com/containerd/ttrpc",
+        "github.com/Microsoft/hcsshim",
```

Comment:
Adding a new dep on google.golang.org/genproto is unexpected ... what genproto packages is hcsshim depending on? It _should_ be able to use isolated submodules like google.golang.org/genproto/googleapis/api and google.golang.org/genproto/googleapis/rpc instead

Reply:
its an indirect require in hcsshim. It comes from containerd/1.7 https://github.com/containerd/containerd/blob/69ae5c026940f806b84c4d1e39d918efccb55b36/go.mod#L77

Reply:
hrmm... but I thought hcsshim dropped the containerd/containerd dep...

Reply:
nope we still need it for quite a few things and we wouldn't be able to drop using containerd/containerd

Reply:
hrmm... if github.com/containerd/containerd@1.7 is still in the graph (which this genproto transitive dep seems like it is...), it seems like we still hit the same issue mentioned in https://github.com/kubernetes/kubernetes/pull/126361#issuecomment-2264232155

> We can't update github.com/containerd/containerd past v1.4.x (so a 2-year-old version) because v1.5.0+ adds a go.mod which surfaces the loopback dependency on k8s.io staging libraries

Reply:
I'm trying to understand why we're picking up *some* transitive dependencies through containerd/containerd but it isn't showing up in our go.mod or triggering recursive k8s.io deps